enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+cannot be implemented externally to the component involved via the
+current SNC build process.
+
+Third, the specific OpenShift component configurations produced via
+the SNC image build process can only be tested in the context of the
+complete CRC product. Since these configurations do not exist within
+the component code repositories, they cannot be tested there, or
+elsewhere, in isolation. This impacts the supportability of CRC
+clusters.
+
+Finally, the single node developer cluster configuration produced by
+the current SNC build process cannot easily be reused in any context
+other than the local VM deployments produced by the CRC
+installer. Similar cluster configurations could be useful in other
+contexts, such as in cloud-based development environments or
+application CI toolchains. Any use case other than for CRC is out of
+scope for this document and would require a separate enhancement PR.
+
+Much of this technical debt associated with the current SNC build
+process can be reduced or eliminated, and better optimization can be
+enabled, by introducing a single-node-developer cluster profile that
+allows each OpenShift component to control how it runs in single node
+developer cluster deployments such as CRC.
+
+### Goals
+
+* Reduce the brittleness of the SNC image build process by moving
+  control of each component’s inclusion and configuration in single
+  node developer clusters from the SNC tooling to the component
+  itself.
+* Reduce CRC resource requirements by allowing better control and
+  optimization of each component for single node developer
+  clusters. This can range from controlling whether or not the
+  component is included in the cluster, through setting configuration
+  details, all the way to adding functionality specific to single node
+  developer clusters. Although the developer tools team may propose
+  patches implementing functionality specific to these clusters, the
+  component teams will be responsible for reviewing and merging such
+  patches and for ongoing maintenance of affected code.
+* Improve supportability by allowing these configurations to be
+  tested, when applicable, within the individual components’ CI
+  processes. Any specialized configuration or features used in single
+  node developer clusters will need to be supported by the component
+  teams, and covered by those components’ unit and CI testing.
+* Decouple the definition of single node developer cluster
+  configuration from the SNC tooling and CRC installer, allowing it to
+  potentially be reused in various other contexts. For example, single
+  node developer clusters could be deployed directly using the
+  standard OpenShift installer or some new specialized installer, or
+  VM images could be created by means other than the current SNC
+  tooling. Any such use cases would require separate enhancement PRs.
+* Ensure that single node developer OpenShift clusters are as
```

Comment:
Please be precise:

Ensure that the CRC use case passes a well-define set of OpenShift conformance tests.  Tests that must be skipped due to topology will be skipped with this profile.